tests/gold_tests/tunnel/tunnel_transform.test.py

```diff
@@ -0,0 +1,168 @@
+tr.StillRunningAfter = server
+
+# Parking this as a ready tester on a meaningless process
+# To stall the test runs that check for the stats until the
+# stats have propagated and are ready to read.
+
+
+def make_done_stat_ready(tsenv):
+    def done_stat_ready(process, hasRunFor, **kw):
+        retval = subprocess.run(
+            "traffic_ctl metric get tunnel_transform.test.done",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            env=tsenv)
+        return b'1' in retval.stdout
+
+    return done_stat_ready
+
+
+# number of sessions/transactions opened and closed are equal
+tr = Test.AddTestRun("Check type errors")
+server2.StartupTimeout = 60
+# Again, here the important thing is the ready function not the server2 process
+tr.Processes.Default.StartBefore(server2, ready=make_done_stat_ready(ts.Env))
+tr.Processes.Default.Command = 'traffic_ctl metric get tunnel_transform.error'
+tr.Processes.Default.ReturnCode = 0
+tr.Processes.Default.Env = ts.Env
+tr.Processes.Default.Streams.All = Testers.ContainsExpression(
+    'tunnel_transform.error 0', 'incorrect statistic return, or possible error.')
+tr.StillRunningAfter = ts
+tr.StillRunningAfter = server
+
+check_input_range = '''
+val=`traffic_ctl metric get tunnel_transform.ua.bytes_sent | cut -d ' ' -f 2; test $val -gt 700
+'''
+
+
+def check_range(path, lo, hi):
+    f = open(path, 'r')
+    content = f.read()
+    values = content.split()
+    f.close()
+    if len(values) == 2:
+        val = int(values[1])
+        return val > lo and val < hi, "Check range", "Out of range"
+    else:
+        return false, "Check range", "Out of range"
+
+
+# Ideally, I'd like to cross check the number of TLS bytes received from UA and
+# received from OS to the curl command.  The debug output lists the number of
+# non record data bytes, but it does not enumerate the number of bytes sent in the records
+# as far as I can tell.  The size of the data record will be different from that plain text
+# size due to padding etc.  Leaving the test with the hard coded values for now.  Hopefully,
+# some bright soul can come along later and make this a better test.
+# Perhaps adding a netcat based test case could do that.
+tr = Test.AddTestRun("Fetch bytes sent")
+tr.Processes.Default.Command = "traffic_ctl metric get tunnel_transform.ua.bytes_sent"
+tr.Processes.Default.ReturnCode = 0
+tr.Processes.Default.Env = ts.Env
+tr.StillRunningAfter = ts
+tr.StillRunningAfter = server
+
+path1 = tr.Processes.Default.Streams.stdout.AbsPath
+
+tr2 = Test.AddTestRun("Check the input bytes sent and fetch outptut bytes sent")
+tr2.Processes.Default.Command = 'traffic_ctl metric get tunnel_transform.os.bytes_sent'
+tr2.Processes.Default.ReturnCode = 0
+tr2.Processes.Default.Env = ts.Env
+tr2.StillRunningAfter = ts
+tr2.StillRunningAfter = server
+tr2.Processes.Default.Streams.stdout = Testers.Lambda(lambda info, tester: check_range(path1, 700, 800))
```

Comment:
I couldn't figure out how to get the current streams stdout fed into the check_range function.  I assume it is available via the info argument, but that appears to be a structure and I couldn't find the methods to get the data stream.  So I'm testing the output by reading the standard output file from the previous test case (a pattern I had used in some tls tests).  @dragon512 or @bneradt I'd appreciate pointers to clean this up.  Would be happy to do that in a later PR.  This logic is functional and verifies that the metric falls within a range.